controllers/tempo/microservices_controller.go

```diff
@@ -143,6 +174,12 @@ func (r *MicroservicesReconciler) getStorageConfig(ctx context.Context, tempo v1
 		return nil, fmt.Errorf("storage secret should contain endpoint and bucket, access_key_id and access_key_secret fields")
 	}
 
+	u, err := url.ParseRequestURI(string(storageSecret.Data["endpoint"]))
```

Comment:
it would be probably make sense to extract the validation of the secret into a separate function